chia/wallet/util/puzzle_compression.py

```diff
@@ -0,0 +1,203 @@
+from typing import Dict, Tuple, List, Union, Any
+
+from chia.types.blockchain_format.program import Program
+from chia.types.blockchain_format.sized_bytes import bytes32
+from chia.util.ints import uint16
+from chia.wallet.util.puzzle_representation import PuzzleRepresentation
+
+# These imports should be temporary
+from blspy import G1Element
+from chia.util.byte_types import hexstr_to_bytes
+from chia.wallet.puzzles import p2_delegated_puzzle_or_hidden_puzzle as standard_puzzle
+from chia.wallet.puzzles.cc_loader import CC_MOD
+from chia.wallet.puzzles.load_clvm import load_clvm
+
+OFFER_MOD = load_clvm("settlement_payments.clvm")
+
+"""
+The following are example classes.
+Ideally, all of our current driver code would be put into a puzzle driver class and registered with the PuzzleCompressor
+structure below.  They must all have three methods implemented:
+- match: Take the result of a uncurry() call and return the relevant list of args that will be put into the
+         PuzzleRepresentation. (The criteria for this list of args is that you should be able to fully construct
+         the puzzle with them, which is the next function)
+- construct: Receive a list of args from a PuzzleRepresentation and construct a full puzzle reveal
+- solve: Receive an opaque piece of JSON, and return a solution reveal.  The reason this argument is an opaque piece
+         of JSON is because the api needs to be the same for all drivers, but obviously all programs will need unique
+         arguments for their solution.  This makes it potentially easy for a UI to craft this as a field on a request
+         which the wallet can blindly hand to the driver it has, and recieve a solution to build the spend with.
+"""
+
+
+class StandardPuzzle:
+    @staticmethod
+    def match(
+        puzzle: Program, compressor: "PuzzleCompressor"
+    ) -> Tuple[bool, List[Union[PuzzleRepresentation, Program]]]:
```

Comment:
Why the `List`?  Why `Tuple[bool, <etc>]` instead of an exception or just the empty list being returned?  If multiple values are actually needed, I would write a quick little class.  This applies multiple places.

Reply:
The idea is that this function returns a boolean representing that it matched the puzzle (useful because the calling logic will probably branch if it didn't).  The list is a list of arguments to be passed to it when it's doing the `construct` call (decompressing)